tests/specs/mcd/verification.k

```diff
@@ -0,0 +1,77 @@
+requires "../lemmas.k"
+requires "bin_runtime.k"
+requires "storage.k"
+
+module HASKELL-HASH2 [symbolic, kore]
+    imports EDSL
+
+    syntax Int ::= hash2(Int, Int) [function]
+ // -----------------------------------------
+endmodule
+
+module VERIFICATION
+    imports LEMMAS
+    imports DSS-BIN-RUNTIME
+    imports DSS-STORAGE
+    imports HASKELL-HASH2
+
+    syntax Bool ::= #notPrecompileAddress ( Int ) [function]
+ // --------------------------------------------------------
+    rule #notPrecompileAddress ( X ) => 0 ==Int X orBool (10 <=Int X andBool #rangeAddress(X))
+
+    // arithmetic/data-structures
+    rule I modInt N => I requires 0 <=Int I andBool I <Int N
+    rule I in (SetItem(I') REST) => I in REST requires I =/=Int I'
+
+    // chop
+    rule chop(hash2(V1, V2) +Int 1) => hash2(V1, V2) +Int 1
+
+    // sizeWordStack/sizeByteArray
+    rule #sizeByteArray(WS ++ WS')                        => #sizeByteArray(WS) +Int #sizeByteArray(WS')
+    rule #sizeByteArray(#padToWidth(32, #asByteStack(I))) => 32 requires 0 <=Int I andBool I <Int pow256
+    rule #sizeByteArray(_WS [ _START .. WIDTH ])          => WIDTH
```

Comment:
requires #range(0 < WIDTH <= #sizeByteArray(WS) -Int START)

Reply:
Actually, AFAICT, all we need is `0 <=Int WIDTH andBool 0 <=Int _START`, and I'm not even sure we need `0 <=Int _START`. But if it passes with both conditions, I think we might as well add them.

Reply:
Removed.